packages/typespec-client-generator-core/design-docs/client.md

```diff
@@ -0,0 +1,379 @@
+- When `access` set to `Access.public`, it means this client could be initialized separately, while `Access.internal` could not. Default to `Access.public` for client and `Access.internal` for sub client.
+- When `accessorAccess` set to `Access.public`, it means this client could be get from parent, while `Access.internal` could not. Default to `undefined` for client and `Access.public` for sub client.
+
+For example:
+```typespec
+@service({
+  title: "Pet Store",
+})
+namespace PetStore {
+  @route("/dogs")
+  interface Dogs {
+    feed(): void;
+    pet(): void;
+  }
+
+  @route("/cats")
+  namespace Cats {
+    op feed(): void;
+    op pet(): void;
+  }
+}
+
+@client({
+  name: "DogsClient",
+  service: PetStore,
+})
+namespace DogsClient {
+  @client
+  interface Feed {
+    feed is PetStore.Dogs.feed;
+  }
+
+  @client
+  interface Pet {
+    pet is PetStore.Dogs.pet;
+  }
+}
+
+@client({
+  name: "CatsClient",
+  service: PetStore,
+})
+namespace CatsClient {
+  @client
+  interface Feed {
+    feed is PetStore.Cats.feed;
+  }
+
+  @client
+  interface Pet {
+    pet is PetStore.Cats.pet;
+  }
+}
+```
+The above tsp gets the two root clients: `DogsClient` and `CatsClient`. 
+All of them has two client accessor method to get sub clients, and with `public` access.
+All of them has two sub clients: `Feed` and `Pet`.
+The initialization method of `Feed` and `Pet` is `internal`.
+
+Another example with customization:
+```typespec
+@service({
+  title: "My Service",
+})
+namespace MyService {
+  interface InnerGroup {
+    op upload(@path blobName: string): void;
+  }
+}
+
+namespace MyCustomizations {
+  model InnerGroupClientOptions {
+    blobName: string;
+  }
+
+  @@clientInitialization(InnerGroup, {parameters: InnerGroupClientOptions, access: Access.public, accessorAccess: Access.internal});
+}
+```
+The above tsp gets client `MyServiceClient` and sub client `InnerGroup`. 
+The `MyServiceClient` has a client accessor method whose response is `InnerGroup`. The method has a parameter named `blobName`. The method is `internal`.
+The `InnerGroup`'s `initialization` model's properties contains a property named `blobName`. The method is `public`.
+The method `upload` no longer has `blobName` parameter, its corresponding operation's parameter `blobName` is mapped to the `blobName` property of `InnerGroup`'s `initialization` model.
+
+2. Move client accessor method out of the normal methods list of client and add a new `clientAccessor` property to put it.
+- The idea is from the feedback that the client access way is too complicated.
+
+3. Consolidate `@client` and `@operationGroup`
+- Deprecate decorator `@operationGroup` and `SdkOperationGroup` type.
+- Current explicitly `@operationGroup` could be migrated to `@client`. If `@client` is nested, then it is a sub-client, will follow previous operation group default logic.
```

Comment:
when we move `@operationGroup` to `@client`, we will move them to a client with init access `internal` and accessor access `public`, right?

Reply:
correct.